themes-default/slim/views/manage_manageSearches.mako

```diff
@@ -1,156 +1,197 @@
-
-                status.append(table);
-                $('.forceSceneExceptionRefresh').removeClass('disabled');
-            };
-
-            /**
-             * Update an element with a spinner gif and a descriptive message.
-             * @param {HTMLElement} spinnerContainer - An element we can use to add the spinner and message to.
-             * @param {string} message - The message to display behind the spinner.
-             * @param {boolean} showSpinner - A boolean to show or not show the spinner (gif).
-             */
-            const updateSpinner = function(spinnerContainer, message, showSpinner) {
-                if (showSpinner) {
-                    message = '<img id="searchingAnim" src="images/loading32' +
-                        MEDUSA.config.themeSpinner + '.gif" height="16" width="16" />&nbsp;' + message;
-                }
-                $(spinnerContainer).empty().append(message);
-            };
-
-            /**
-             * Trigger the force refresh of all the exception types.
-             */
-            $('.forceSceneExceptionRefresh').on('click', () => {
-                const status = $('#sceneExceptionStatus');
-                // Start a spinner.
-                updateSpinner(status, 'Retrieving scene exceptions...', true);
-
-                api.post('alias-source/all/operation', { type: 'REFRESH' }, {
-                    timeout: 60000
-                }).then(response => {
-                    status[0].innerHTML = '';
-                    status.append($('<span></span>').text(response.data.result));
-
-                    api.get('alias-source').then(response => {
-                        updateExceptionTable(response.data);
-                        $('.forceSceneExceptionRefresh').addClass('disabled');
-                    }).catch(err => {
-                        log.error('Trying to get scene exceptions failed with error: ' + err);
-                        updateSpinner(status, 'Trying to get scene exceptions failed with error: ' + err, false);
-                    });
-                    updateSpinner(status, 'Finished updating scene exceptions.', false);
-                }).catch(err => {
-                    log.error('Trying to update scene exceptions failed with error: ' + err);
-                    updateSpinner(status, 'Trying to update scene exceptions failed with error: ' + err, false);
+        spinnerSrc() {
+            const { config } = this;
+            const { themeName } = config;
+            const themeSpinner = themeName === 'dark' ? '-dark' : '';
+            return 'images/loading32' + themeSpinner + '.gif';
+        }
+    },
+    methods: {
+        /**
+         * Trigger the force refresh of all the exception types.
+         */
+        forceSceneExceptionRefresh() {
+            const { updateExceptionData, sceneRefresh } = this;
+            // Start a spinner.
+            sceneRefresh.showSpinner = true;
```

Comment:
Should be using `this.` for setting vars, not updating the const.

Reply:
Are you sure? `sceneRefresh` is an object. I've been using this technique right from the start.

Reply:
Yeah this also feels like an edge case to me. Should we use let when changing mutables?

Reply:
Since it's updating the vue data object it shouldn't be using let or const. It should be using `this.`